### DISCUSSION TITLE
Add debug_blacklist by default in laravel

### DISCUSSION THREAD

**Discussion by ghost** [laravel/framework#50600] [Ideas]
'debug_blacklist' => [
        '_COOKIE' => \array_keys($_COOKIE),
        '_SERVER' => \array_keys($_SERVER),
        '_ENV' => \array_keys($_ENV),
    ],

This is a step that I have to do manually on each new project...

**Comment by ahinkle** [comment]
Can you further elaborate on reasons why you would want to do this? Not real security risk without it unless you're deliberately printing these across your app.

**Reply by ghost** [reply]
Many devops turn debug on on public environments => full disclosure of secrets.

**Reply by ahinkle** [reply]
You should never under any circumstance turn on debug mode in public environments.

**Reply by ghost** [reply]
Tell that to them. They come from outside laravel community, see debug and they enable it. It happened to me with multiple devops...

Also the env get prinded without taking into accout APP_ENV being production so that is why I asked for this.
UPDATE I also was witness to other public websites having the same issue.

**Reply by a-h-abid** [reply]
You should make your own starter kit.

**Reply by ghost** [reply]
I did. But laravel gets updated too quickly and is hard to keep up. Anyway laravel/lumen 8 have the least issues from what I experienced and saw in here... I would vote for that to become a LTS.

**Reply by ahinkle** [reply]
The problems you are experiencing are not best practices. I would consult with your internal team to adapt modern practices. Lumen and Laravel 8 are EOL. Laravel does not get updated too quickly- it is just a major release once a year.

**Reply by ghost** [reply]
Best practices can lead to bankruptcy if you believe all you are told ;) 
UPDATE
You can give me 1000000000 thumb downs . It will not change the reality. I saw it happen. Unit tests and refactoring/updating when not needed are the main culprits for the bankruptcies I saw happen.
